hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/management/actionhistory/ActionStatusMsgGrid.java

```diff
@@ -0,0 +1,210 @@
+/**
+ * Copyright (c) 2015 Bosch Software Innovations GmbH and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.hawkbit.ui.management.actionhistory;
+
+import java.util.Map;
+
+import org.eclipse.hawkbit.ui.common.grid.AbstractGrid;
+import org.eclipse.hawkbit.ui.utils.SPUIDefinitions;
+import org.eclipse.hawkbit.ui.utils.SPUIStyleDefinitions;
+import org.eclipse.hawkbit.ui.utils.UIComponentIdProvider;
+import org.eclipse.hawkbit.ui.utils.VaadinMessageSource;
+import org.vaadin.addons.lazyquerycontainer.BeanQueryFactory;
+import org.vaadin.addons.lazyquerycontainer.LazyQueryContainer;
+import org.vaadin.addons.lazyquerycontainer.LazyQueryDefinition;
+import org.vaadin.spring.events.EventBus.UIEventBus;
+
+import com.google.common.collect.Maps;
+import com.vaadin.data.Item;
+import com.vaadin.event.ItemClickEvent;
+import com.vaadin.event.ItemClickEvent.ItemClickListener;
+import com.vaadin.ui.Component;
+import com.vaadin.ui.TextArea;
+import com.vaadin.ui.themes.ValoTheme;
+
+/**
+ * This grid presents the messages for a selected action-status.
+ */
+public class ActionStatusMsgGrid extends AbstractGrid<LazyQueryContainer> {
+    private static final long serialVersionUID = 1L;
+
+    private static final String[] rightAlignedColumns = new String[] { ProxyMessage.PXY_MSG_ID };
+
+    private final String noMsgText;
+
+    private final AlignCellStyleGenerator alignGenerator;
+
+    private final BeanQueryFactory<ActionStatusMsgBeanQuery> targetQF = new BeanQueryFactory<>(
+            ActionStatusMsgBeanQuery.class);
+
+    /**
+     * Constructor.
+     *
+     * @param i18n
+     * @param eventBus
+     */
+    protected ActionStatusMsgGrid(final VaadinMessageSource i18n, final UIEventBus eventBus) {
+        super(i18n, eventBus, null);
+        noMsgText = createNoMessageProxy(i18n);
+
+        setSingleSelectionSupport(new SingleSelectionSupport(true));
+        setDetailsSupport(new DetailsSupport());
+
+        alignGenerator = new AlignCellStyleGenerator(null, null, rightAlignedColumns);
+        addStyleName(SPUIStyleDefinitions.ACTION_HISTORY_MESSAGE_GRID);
+
+        setDetailsGenerator(new MessageDetailsGenerator());
+
+        this.addItemClickListener(new ItemClickListener() {
+            private static final long serialVersionUID = 1L;
+
+            @Override
+            public void itemClick(final ItemClickEvent event) {
+                final Object itemId = event.getItemId();
+                setDetailsVisible(itemId, !isDetailsVisible(itemId));
+            }
+        });
+
+        init();
```

Comment:
Constructor should not call overridables see: http://www.javapractices.com/topic/TopicAction.do?Id=215